### PR TITLE
fix(discussions): make sure container exists before checking permissions

### LINKED ISSUE #9383: discussion/add fatal error
`discussion/add` doesn't properly handle container guid before calling `canWriteToContainer()`

### PR DESCRIPTION
**Pull request by hypeJunction**
Lacking or non existent guid resulted in Fatal Error

Fixes #9383

### CODE REVIEW

File: mod/discussions/views/default/resources/discussion/add.php [hunk 1]
```diff
@@ -3,6 +3,9 @@
 elgg_gatekeeper();
 
 $guid = elgg_extract('guid', $vars);
+
+elgg_entity_gatekeeper($guid);
```
Comment (hypeJunction) [inline mod/discussions/views/default/resources/discussion/add.php]: Do we also need elgg_group_gatekeeper()?

Comment (mrclay) [inline mod/discussions/views/default/resources/discussion/add.php]: Probably yes.

Comment (hypeJunction) [inline mod/discussions/views/default/resources/discussion/add.php]: Done

**Comment by mrclay** [comment]
LGTM

**Pull request merged**